src/APIBasedAuth.ts

```diff
@@ -125,23 +182,37 @@ declare namespace APIBasedAuth {
     session?: string;
   };
 
-  export type Session = {
-    session: string;
-    username: string;
-  };
+  type Session = Record<SessionKey, string>;
+
+  type SessionKey = 'session' | 'username';
 
   export type SignInData = {
     clientId: string;
     password: string;
     username: string;
   };
 
-  export type SuccessfulAuthResponse = {
-    accessToken: string;
-    idToken: string;
-    refreshToken: string;
+  export type Storage = {
+    getItem(key: StorageKeys[StorageKey]): string | Promise<string>;
```

Comment:
Could you simplify this to always return a `Promise`?

Reply:
@mdlavin if we do that, the default window.localStorage does not fit into the storage type as the `get/set/remove` within that aren't promises

Reply:
That's a bummer. I won't be the one to maintain it, but I'd be tempted to wrap `window.localStorage` to have a Promise API just for sake of consistency.